openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from datetime import datetime
+from urllib.parse import quote_plus
+
+import requests
+from django.conf import settings
+from rest_framework import status
+
+from openedx_external_enrollments.edxapp_wrapper.get_site_configuration import configuration_helpers
+from openedx_external_enrollments.edxapp_wrapper.get_student import get_user
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import EnrollmentRequestLog
+
+LOG = logging.getLogger(__name__)
+
+
+class MITHzInstanceExternalEnrollment(BaseExternalEnrollment):
+    """
+    MITHzInstanceExternalEnrollment class.
+    """
+
+    def __init__(self):
+        self.MIT_HZ_PROVIDER = configuration_helpers.get_value(
+            'MIT_HZ_PROVIDER',
+            'samlp',
+        )
+        self.MIT_HZ_ORG = configuration_helpers.get_value(
+            'MIT_HZ_ORG',
+            'Pearson',
+        )
+
+    def __str__(self):
+        return 'mit_hz'
+
+    def _get_bearer_token(self):
+        """
+        Returns a bearer token required for the Authorization header.
+        """
+        url = '{root_url}{path}'.format(
+            root_url=settings.MIT_HZ_API_URL,
+            path=settings.MIT_HZ_LOGIN_PATH,
+        )
+        data = {
+            'id': settings.MIT_HZ_ID,
+            'secret': settings.MIT_HZ_SECRET,
+        }
+        log_details = {
+            'url': url,
+        }
+        try:
+            response = requests.post(
+                url=url,
+                json=data,
+            )
+        except Exception as error:
+            msg_error = 'Failed to authenticate at MIT HZ API. Reason: %s'
+            log_details['response'] = {'error': msg_error + str(error)}
+            LOG.error(msg_error + str(error))
```

Comment:
Can we concatenate the error message in a single statement? @luismorenolopera 
```suggestion
            msg_error = 'Failed to authenticate at MIT HZ API. Reason: %s' % str(error)
            log_details['response'] = {'error': msg_error}

            LOG.error(msg_error)
```